libs/substrate-runtime-proposal-hash/src/lib.rs

```diff
@@ -116,8 +116,8 @@ mod prop_hash_tests {
 		assert_eq!(
 			get_call_hash(PREFIX_SYSTEM_SETCODE, &[1, 2, 42]).unwrap(),
 			[
-				156, 244, 243, 93, 21, 8, 113, 238, 186, 17, 20, 52, 240, 236, 140, 15, 108, 26, 86, 5, 152, 148, 91,
-				162, 108, 168, 3, 65, 254, 162, 114, 46
+				174, 123, 79, 240, 254, 106, 147, 26, 65, 61, 41, 84, 181, 102, 24, 182, 128, 135, 188, 31, 135, 187,
+				99, 34, 143, 35, 120, 100, 246, 90, 186, 106
```

Comment:
Hm I just tested this in the kusama runtime and it returns this:  
```rust
#[test]
fn encode_test() {
	let call = frame_system::Call::<Runtime>::set_code {
		code: vec![1, 2, 42],
	};
	let hash = sp_core::hashing::blake2_256(&call.encode());
	panic!("{:?}", hash);
}
```
`[148, 109, 109, 44, 135, 214, 234, 93, 108, 170, 48, 66, 109, 234, 165, 185, 35, 178, 164, 58, 172, 159, 112, 176, 66, 254, 114, 31, 64, 22, 217, 38]`.  
But I guess it works if you tested it with the real runtime.

Reply:
I did not merge yet because did not have time to fully test yet. 

What do you mean with:
> tested this in the kusama runtime

Reply:
The test above does not consider any real runtime (neither Kusama nor another). The tested "runtime" is litterally `[1, 2, 42]` and probably does not do much :)